subsys/mgmt/hawkbit: change 'Hawkbit' & 'hawkbit' to 'hawkBit'

ycsin opened the pull request:
Change 'Hawkbit' and 'hawkbit' to 'hawkBit' wherever makes sense.

[preview](https://builds.zephyrproject.io/zephyr/pr/71057/docs/samples/subsys/mgmt/hawkbit/README.html)

ycsin committed:
subsys/mgmt/hawkbit: change 'Hawkbit' & 'hawkbit' to 'hawkBit'

Change 'Hawkbit' and 'hawkbit' to 'hawkBit' wherever
makes sense, and a bit of touch ups here and there.

Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

maass-hamburg reviewed (dismissed):
LGTM

kartben commented on samples/subsys/mgmt/hawkbit/README.rst:
```diff
@@ -25,22 +25,22 @@ source code for this mode hawkbit_manual
 *******
 
-* The Zephyr port of ``Hawkbit`` is configured to run on a
+* The Zephyr port of ``hawkBit`` is configured to run on a
```
I know you only did the search replace, but I don't think it makes sense to use double quotes here (and several other instances in this file)

<img width="603" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/a3f6080e-cab6-4b95-8fd1-29e7f4ccf8eb">

kartben replied:
@ycsin would be great to address this as part of the clean-up?

ycsin replied:
Yeah I forgot about this, will do another rev

kartben commented on samples/subsys/mgmt/hawkbit/prj.conf:
```diff
@@ -18,7 +18,7 @@ CONFIG_BOOTLOADER_MCUBOOT=y
 CONFIG_MAIN_STACK_SIZE=4096
 
-#Minimal Heap mem pool size for the Hawkbit working
+#Minimal Heap mem pool size for the hawkBit working
```
```suggestion
#Minimal Heap mem pool size for hawkBit to work
```

kartben commented on subsys/mgmt/hawkbit/Kconfig:
```diff
@@ -2,7 +2,7 @@
 
 menuconfig HAWKBIT
-	bool "Hawkbit Firmware Over-the-Air support"
+	bool "hawkBit Firmware Over-the-Air support"
```
albeit a bit tricky, let's try to comply with the Eclipse Foundation Trademark Usage Policy and change this one as follows:

```suggestion
	bool "Eclipse hawkBit Firmware Over-the-Air support"
```

kartben reviewed (changes requested):
thanks for doing this!

kartben commented on samples/subsys/mgmt/hawkbit/prj.conf:
```diff
@@ -1,4 +1,4 @@
-#Turn on the Hawkbit
+#Turn on the hawkBit
```
```suggestion
#Turn on hawkBit
```

kartben commented on subsys/mgmt/hawkbit/hawkbit_priv.h:
```diff
@@ -7,7 +7,7 @@
  *
  * @brief This file contains structures representing JSON messages
- * exchanged with a hawkbit
+ * exchanged with a hawkBit
```
```suggestion
 * exchanged with a hawkBit server
```

ycsin commented:
Thanks @kartben, good catch!

ycsin commented on subsys/mgmt/hawkbit/hawkbit.c:
```diff
@@ -10,7 +10,7 @@
 #include <zephyr/logging/log.h>
 
-LOG_MODULE_REGISTER(hawkbit, CONFIG_HAWKBIT_LOG_LEVEL);
+LOG_MODULE_REGISTER(hawkBit, CONFIG_HAWKBIT_LOG_LEVEL);
```
sure, I also forgot to remove the double tick, will do another rev

ycsin committed:
samples: mgmt: hawkbit: make references to "frdm_k64f" consistent

The "frdm_k64f" is referred as frdm_k64f and ``frdm_k64f`` in
places, update them to ":ref:`Freedom-K64F <frdm_k64f>`" or
``frdm_k64f``.

Signed-off-by: Yong Cong Sin <yongcong.sin@gmail.com>

ycsin commented:
@kartben I also made some changes to the 'frdm_k64f', LMK what you think, thanks

kartben reviewed (approved):
nice - thanks @ycsin!
